view/Requirements.php

```diff
@@ -1439,16 +1464,24 @@ function() use ($fileList, $minify, $type) {
 				}
 			);
 
-		// Since url won't be automatically suffixed, add it in here
-		if($this->getSuffixRequirements()) {
-			$q = stripos($combinedURL, '?') === false ? '?' : '&';
-			$combinedURL .= "{$q}m={$entropy}";
-		}
-
 		return $combinedURL;
 	}
 
 	/**
+	 * Given a filename and list of files, generate a new filename unique to these files
+	 *
+	 * @param string $name
+	 * @param array $files
+	 * @return string
+	 */
+	protected function hashedCombinedFilename($combinedFile, $fileList) {
+		$name = pathinfo($combinedFile, PATHINFO_FILENAME);
+		$hash = $this->hashOfFiles($fileList);
+		$extension = File::get_file_extension($combinedFile);
+		return $name . '-' . substr($hash, 0, 7) . '.' . $extension;
```

Comment:
Changing from `myfile.js?m=<hash>` to `myfile-<hash>.js` introduces synchronisation issues on distributed environments which are using `csync` on local filesystems rather than a centralised store like S3. A request to `admin/pages` might be served by instance1, including `<script src="myfile-<hash>.js">`. The browser then requests `myfile-<hash>.js` from instance2 shortly afterwards (<1s). Depending on `csync` latency, this file might not exist there yet. This problem was less likely with `myfile.js?m=<hash>` since the "old" file was served. Of course this could lead to its own issues if the generated HTML doesn't match the served JS or CSS, but at least it didn't crash the CMS completely.

`csync` will become a less common approach in 4.0, but this changeset makes its use more fragile, right?

/cc @caffeineinc